eclipse-collections/src/main/java/org/eclipse/collections/impl/set/mutable/AbstractMultiReaderSet.java

```diff
@@ -0,0 +1,1004 @@
+    }
+
+    @Override
+    public MutableSet<T> without(T element)
+    {
+        this.remove(element);
+        return this;
+    }
+
+    @Override
+    public MutableSet<T> withAll(Iterable<? extends T> elements)
+    {
+        this.addAllIterable(elements);
+        return this;
+    }
+
+    @Override
+    public MutableSet<T> withoutAll(Iterable<? extends T> elements)
+    {
+        this.removeAllIterable(elements);
+        return this;
+    }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if (o == this || o == this.delegate)
+        {
+            return true;
+        }
+
+        if (!(o instanceof Set))
+        {
+            return false;
+        }
+
+        try (LockWrapper wrapper = this.lockWrapper.acquireReadLock())
+        {
+            return this.delegate.equals(o);
+        }
+    }
+
+    @Override
+    public int hashCode()
+    {
+        try (LockWrapper wrapper = this.lockWrapper.acquireReadLock())
+        {
+            return this.delegate.hashCode();
+        }
+    }
+
+    // Exposed for testing
+
+    static final class UntouchableMutableSet<T>
+            extends UntouchableMutableCollection<T>
+            implements MutableSet<T>
+    {
+        private final MutableList<UntouchableIterator<T>> requestedIterators = mList();
+
+        private UntouchableMutableSet(MutableSet<T> newDelegate)
+        {
+            this.delegate = newDelegate;
+        }
+
+        public void becomeUseless()
+        {
+            this.delegate = null;
+            this.requestedIterators.each(UntouchableIterator::becomeUseless);
+        }
+
+        /**
+         * This must be implemented this way to guarantee a reference to the delegate doesn't escape.
+         */
+        @Override
+        public MutableSet<T> with(T element)
+        {
+            this.add(element);
+            return this;
+        }
+
+        @Override
+        public MutableSet<T> without(T element)
+        {
+            this.remove(element);
+            return this;
+        }
+
+        @Override
+        public MutableSet<T> withAll(Iterable<? extends T> elements)
+        {
+            this.addAllIterable(elements);
+            return this;
+        }
+
+        @Override
+        public MutableSet<T> withoutAll(Iterable<? extends T> elements)
+        {
+            this.removeAllIterable(elements);
+            return this;
+        }
+
+        @Override
+        public MutableSet<T> asSynchronized()
+        {
+            throw new UnsupportedOperationException("Cannot call asSynchronized() on " + this.getClass().getSimpleName());
+        }
+
+        @Override
+        public MutableSet<T> asUnmodifiable()
+        {
+            throw new UnsupportedOperationException("Cannot call asUnmodifiable() on " + this.getClass().getSimpleName());
+        }
+
+        @Override
+        public ImmutableSet<T> toImmutable()
+        {
+            return Sets.immutable.withAll(this.getDelegate());
```

Comment:
These can all just be `delegate` instead of `this.getDelegate()` if we need, right?

Reply:
yes